### PR TITLE
feat(dotnet): Open Telemetry support

### PR DESCRIPTION
**Pull request by bitsandfoxes**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

*Describe your changes here. If your PR relates to or resolves an issue, add a link to that too.*

### CODE REVIEW

**Review by AbhiPrasad — APPROVED** [review]

File: src/platform-includes/performance/opentelemetry-install/dotnet.mdx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+Install-Package Sentry -Version {{@inject packages.version('sentry.dotnet') }}
+Install-Package Sentry.OpenTelemetry -Version {{@inject packages.version('sentry.dotnet') }}
+```
+
+```shell {tabTitle:Paket CLI}
+paket add Sentry --version {{@inject packages.version('sentry.dotnet') }}
+paket add Sentry.OpenTelemetry --version {{@inject packages.version('sentry.dotnet') }}
+```
+
+If you're building an ASP.NET or ASP.NET Core application you'll want to add their respective packages as well, i.e. `Sentry.AspNet` or `Sentry.AspNetCore`.
```
Comment (vivianyentran) [inline src/platform-includes/performance/opentelemetry-install/dotnet.mdx:24]:
```suggestion
If you're building an ASP.NET or ASP.NET Core application, add their respective packages (`Sentry.AspNet` or `Sentry.AspNetCore`) as well.
```

File: src/platform-includes/performance/opentelemetry-setup/dotnet.mdx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    .AddSource(serviceName)
+    .ConfigureResource(resource =>
+        resource.AddService(
+            serviceName: serviceName,
+            serviceVersion: serviceVersion))
+    .AddSentry() // <-- Configure OpenTelemetry to send traces to Sentry
+    .Build();
+```
+
+Next, initialize Sentry and opt into the use of OpenTelemetry. This allows the SDK to sent OpenTelemetry spans to be sent to Sentry.
```
Comment (vivianyentran) [inline src/platform-includes/performance/opentelemetry-setup/dotnet.mdx:16]:
```suggestion
Next, initialize Sentry and opt into the use of OpenTelemetry. This allows the SDK to send OpenTelemetry spans to Sentry.
```

File: src/platform-includes/performance/opentelemetry-setup/dotnet.mdx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        tracerProviderBuilder
+            .AddSource(Telemetry.ActivitySource.Name)
+            .ConfigureResource(resource => resource.AddService(Telemetry.ServiceName))
+            .AddAspNetCoreInstrumentation()
+            .AddHttpClientInstrumentation()
+            .AddSentry() // <-- Configure OpenTelemetry to send trace information to Sentry
+    );
+```
+
+Next, initialize Sentry and opt into the use of OpenTelemetry. This allows the SDK to sent OpenTelemetry spans to be sent to Sentry.
```
Comment (vivianyentran) [inline src/platform-includes/performance/opentelemetry-setup/dotnet.mdx:45]:
```suggestion
Next, initialize Sentry and opt into the use of OpenTelemetry. This allows the SDK to send OpenTelemetry spans to Sentry.
```

File: src/platform-includes/performance/opentelemetry-setup/dotnet.mdx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    .AddAspNetInstrumentation()
+    .AddSource(Telemetry.ServiceName)
+    .SetResourceBuilder(
+        ResourceBuilder.CreateDefault()
+            .AddService(serviceName: Telemetry.ServiceName, serviceVersion: "1.0.0")
+        );
+
+```
+
+Next, initialize Sentry and opt into the use of OpenTelemetry. You'll need to provide the SDK with the builder for OpenTelemetry's tracer provider to allow sending spans to Sentry.
```
Comment (vivianyentran) [inline src/platform-includes/performance/opentelemetry-setup/dotnet.mdx:71]:
```suggestion
Next, initialize Sentry and opt into the use of OpenTelemetry. Provide the SDK with the builder for OpenTelemetry's tracer provider to allow sending spans to Sentry.
```

**Review by vivianyentran — APPROVED** [review]
Thanks for updating the docs!

File: src/platform-includes/performance/opentelemetry-setup/dotnet.mdx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+{
+    options.Dsn = "...Your DSN...";
+    options.TracesSampleRate = 1.0;
+    options.UseOpenTelemetry(); // <-- Configure Sentry to use OpenTelemetry trace information
+});
+```
+
+### ASP.NET applications
+
+To start tracing in an ASP.NET application, you need to create a tracer provider.
```
Comment (lizokm) [inline src/platform-includes/performance/opentelemetry-setup/dotnet.mdx:58]:
```suggestion
To start tracing in an ASP.NET application, you'll need to create a tracer provider.
```

**Review by lizokm — APPROVED** [review]
Small language tweaks.

**Comment by bitsandfoxes** [comment]
Thanks a lot! <3

**Pull request merged**